api/v1alpha1/warehouse_types.go

```diff
@@ -124,7 +124,7 @@ type GitSubscription struct {
 	// subscription is implicitly to the repository's default branch.
 	//
 	// +kubebuilder:validation:MinLength=1
-	// +kubebuilder:validation:Pattern=`^\w+([-/]\w+)*$`
+	// +kubebuilder:validation:Pattern=`^\w+([-\/]\w+)*(-\d+\.\d+)?$`
```

Comment:
I'm supportive of relaxing this, but I don't think it's relaxed _enough_.

There are a lot of valid branch names containing a `.` that this pattern will not match.

`foo.bar` should be acceptable, for instance.

Also, `/` needs to be escaped in PCRE, but not in Go.

I suggest this:

```suggestion
	// +kubebuilder:validation:Pattern=`^\w+([-/\.]\w+)*$`
```

https://regex101.com/r/J7GXoJ/1

Codegen will need to be re-run.

Reply:
I will extend the regex to be more relaxed 😎

Reply:
Thanks @tal-hason! Appreciate this!